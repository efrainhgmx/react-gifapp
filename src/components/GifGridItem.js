import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, url, title } ) => {

    console.log( id, url, title )
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propsTypes = {
    id: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
