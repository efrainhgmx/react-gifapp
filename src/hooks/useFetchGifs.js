import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    //Recibe Dos Argumentos,
    //1.- una función  que es la función a ejecutar
    //2.- Un arreglo de dependencias, si se envia un arreglo vacio, la función solo se ejecuta una vez.
    useEffect( () => {
        getGifs( category )
            .then( imgs => {

                setTimeout( () => {
                    setState({
                        data: imgs,
                        loading: false,
                    })
                }, 3000)

            } );
    }, [ category ]); 


    return state; //{ data: [], loading: true }

}
