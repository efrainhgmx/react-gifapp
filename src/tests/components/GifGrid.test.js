import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test on <GifGrid /> component', () => {
    

    const category = "Batman";
    
    test('Snapshot of component', () => {
        const wrapper = shallow(<GifGrid category= { category }/>);
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should be show items when the load is done useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com/google.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category= { category }/>);


        expect(wrapper.find('p').exists()).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
    
});
