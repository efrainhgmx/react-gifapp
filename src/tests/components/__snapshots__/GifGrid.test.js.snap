// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGrid /> component Snapshot of component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Batman
  </h3>
  <p>
    Cargando....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
